### PR TITLE
custom npm argument for environment

### PR DESCRIPTION
**Pull request by manuelcorrea**

### CODE REVIEW

**Comment by rafibomb** [comment]
I think this is a great change! It is a breaking change so we can push it to the next point release 6.3.
Are you able to point this change to the v6.3 branch so we can merge it in?

**Comment by manuelcorrea** [comment]
@rafibomb it is already pointing to the 6.3 branch. Thanks!

**Comment by phifa** [comment]
What does this actually do? The former ...`build` is now `build --production`? What is `build` now?

**Comment by manuelcorrea** [comment]
@phifa It will be default, which I guess in this case is non-environment = development?

In our internal project we ended with 

```
"scripts": {
    "start": "gulp",
    "build": "gulp build --production",
    "build-production": "gulp build --production",
    "build-dev": "gulp build --development"
  },
```

Production will be the default and if you want the raw files for debugging then do a build-dev?

I didn't want to push this change to favor more flexibility on the script environment through cli.

Thoughts?

**Comment by manuelcorrea** [comment]
Alternative the default can be set on the gulp file instead of the package.json ...

File: readme.md [hunk 1]
```diff
@@ -62,4 +62,4 @@ Finally, run `npm start` to run Gulp. Your finished site will be created in a fo
 http://localhost:8000
 ```
 
-To create compressed, production-ready assets, run `npm run build`.
+To create compressed, production-ready assets, run `npm run build -- --production`.
```
Comment (colin-marshall) [inline readme.md:65]: There are two pairs of hyphens here. I believe this is a typo.

**Review by colin-marshall — CHANGES_REQUESTED** [review]

File: readme.md (hunk 1 above)
Comment (manuelcorrea) [reply readme.md:65]: Hey Colin, it is not a typo. This is the way to pass arguments to npm scripts 
https://docs.npmjs.com/cli/run-script

```
As of npm@2.0.0, you can use custom arguments when executing scripts. The special option -- is used by getopt to delimit the end of the options. npm will pass all the arguments after the -- directly to your script:

npm run test -- --grep="pattern"
The arguments will only be passed to the script specified after npm run and not to any pre or post script.
```

Comment (colin-marshall) [reply readme.md:65]: @manuelcorrea good to know, thanks!

**Pull request closed without merging**